packages/next/src/server/base-server.ts

```diff
@@ -1955,9 +1955,11 @@ export default abstract class Server<
      * prefetch request.
      */
     const isPrefetchRSCRequest =
-      (req.headers[NEXT_ROUTER_PREFETCH_HEADER.toLowerCase()] === '1' ||
```

Comment:
is `isPrefetchRSCRequest` guaranteed to have been set by the time we hit this block?

Reply:
In this case yes, the vary header is only called after a render is performed (and we have a resolved pathname, and know it's an app path or not) because it's assigned in the request handler.